### PR TITLE
Docs - sidebar headings & toggles

### PR DESCRIPTION
**Pull request by samrichca**
- update the left sidebar:
  - toggle submenus open/closed
  - convert top level folder titles to H4 tags
  - persist user interaction with this component across pages
  - styling changes
- other changes indicated in comments

### CODE REVIEW

File: src/hooks/useActiveId.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ import { useEffect, useState } from 'react';
 
 // ref: https://nickymeuleman.netlify.app/blog/table-of-contents#get-the-active-headings-id
 
-export default function useActiveId(itemIds) {
+export default function useActiveId(itemIds = []) {
```
Comment (samrichca) [inline src/hooks/useActiveId.js:8]: this component was crashing when navigating between docs pages or updating a page with hot reloading in development (i.e. writing markdown content)

so I added handling for no itemIds & "missing" elements

File: src/templates/documentation/index.js [hunk 2]
```diff
@@ -254,6 +202,10 @@ export const pageQuery = graphql`
             pages {
               title
               url
+              pages {
```
Comment (samrichca) [inline src/templates/documentation/index.js:205]: forgot to query a navigation level :)

File: src/templates/documentation/index.js [hunk 3]
```diff
@@ -98,130 +93,83 @@ export default function DocumentationPage({ data, location, path }) {
...
-                  </div>
+            <React.Fragment>
+              <MDXProvider
+                components={{
+                  ...replacedComponents,
+                  ...shortcodes,
+                  a: props => <LinkHelper {...props} location={location} />,
+                  // the page title is h1
+                  // so demote markdown headings by one level
+                  h1: props => <AnchorHeading location={location} size="h2" {...props} />,
```
Comment (anncatton) [inline src/templates/documentation/index.js:117]: Can do with an array map for each header type maybe?

**Review by anncatton — COMMENTED** [review]
just one small suggestion :)

File: src/templates/documentation/index.js (hunk 3 above)
Comment (samrichca) [reply src/templates/documentation/index.js:117]: could do!

**Review by anncatton — APPROVED** [review]

**Pull request merged**